src/transformers/utils/quantization_config.py

```diff
@@ -418,3 +429,61 @@ def post_init(self):
                     f"""dataset needs to be either a list of string or a value in
                     ['wikitext2','c4','c4-new','ptb','ptb-new'], but we found {self.dataset}"""
                 )
+
+
+@dataclass
+class AwqConfig(QuantizationConfigMixin):
+    """
+    This is a wrapper class about all possible attributes and features that you can play with a model that has been
+    loaded using `auto-awq` library awq quantization relying on auto_awq backend.
+
+    Args:
+        bits (`int`):
+            The number of bits to quantize to.
+        group_size (`int`, *optional*, defaults to 128):
+            The group size to use for quantization. Recommended value is 128 and -1 uses per-column quantization.
+        zero_point (`bool`, *optional*, defaults to `True`):
+            Whether to use zero point quantization.
+        version (`str`, *optional*, defaults to `"GEMM"`):
+            The version of the quantization algorithm to use.
+        backend (`str`, *optional*, defaults to `"autoawq"`):
+            The quantization backend. Some models might be quantized using `llm-awq` backend. This is useful for users
+            that quantize their own models using `llm-awq` library.
+    """
+
+    def __init__(
+        self,
+        bits: int,
```

Comment:
You can make this optional too with a default value to 4. If in the future, we want to change a config related to the kernels, the user won't have to set this arg